indico/modules/rb/models/reservations.py

```diff
@@ -411,6 +411,12 @@ def get_with_data(*args, **kwargs):
 
         return result.values()
 
+    def is_equal_with(self, new_booking):
```

Comment:
The name is too generic (it sounds like a poor man's `__eq__`), so something like `has_same_dates` would be more suitable.

But maybe it's a bit too simple to justify a method in the model unless we use this in multiple places. And even then, do we have it in a dict in both places? Not sure how the code for splitting looks like right now.
I'd probably simply check it in the controller and eventually move it into an operation function.